beacon-chain/state/state-native/v2/state_trie.go

```diff
@@ -30,27 +28,6 @@ func InitializeFromProto(st *ethpb.BeaconStateAltair) (state.BeaconStateAltair,
 	return InitializeFromProtoUnsafe(proto.Clone(st).(*ethpb.BeaconStateAltair))
 }
 
-// InitializeFromSSZReader can be used when the source for a serialized BeaconState object
-// is an io.Reader. This allows client code to remain agnostic about whether the data comes
-// from the network or a file without needing to read the entire state into mem as a large byte slice.
-func InitializeFromSSZReader(r io.Reader) (state.BeaconStateAltair, error) {
```

Comment:
This method uses `InitializeFromProtoUnsafe` under the hood, which performs the flag check. Does that cover the concern, or is there a reason why every method needs to be mirrored?

Reply:
I think I get it, the methods are being mirrored in this way because you want to eventually repoint code calling the `state/vN` methods to `state/native-state/vN`. I think I'm going to just move these methods out of the state package to steer clear of these native state changes. In the final PR they are only used in the `detect` package, so I think that would be a fine home for them.